src/producer.rs

```diff
@@ -254,20 +257,19 @@ impl<P: Partitioner> Producer<P> {
               V: AsBytes
     {
         let mut rs = try!(self.send_all(ref_slice(rec)));
-        if self.config.required_acks == 0 {
+
+        if self.config.required_acks != 0 {
             // ~ with no required_acks we get no response and
             // consider the send-data request blindly as successful
-            Ok(())
-        } else {
             assert_eq!(1, rs.len());
-            rs.pop().unwrap().offset.map(|_| ())
+            rs.pop().unwrap();
         }
+
+        Ok(())
```

Comment:
Ohh, I see now. I'm sorry, I think I just had a brain fart when I read this function and didn't see what the `map(|_| ())` was doing. I'll fix this.